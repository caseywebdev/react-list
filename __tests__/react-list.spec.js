import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import ReactTestRenderer from 'react-test-renderer';
import ReactList from './../react-list';

describe('react-list', function () {

  const createNodeMock = function createNodeMock(element) {
    if (element.type === 'div') {
      // div container of react-list
      return document.createElement('div');
    }
    if (element.type === 'ul') {
      // element of the <UniformList>
      return document.createElement('ul');
    }
    // You can return any object from this method for any type of DOM component.
    // React will use it as a ref instead of a DOM node when snapshot testing.
    return null;
  };

  const render = function render(component) {
    const options = {
      createNodeMock
    };
    return ReactTestRenderer.create(component, options);
  };

  class MyList extends React.Component {
    constructor(props) {
      super(props);
      this.listRenderer = this.listRenderer.bind(this);
      this.listItemRenderer = this.listItemRenderer.bind(this);
    }

    listItemRenderer(index, key) {
      return (
        <li key={key}>
          list item: {this.props.items[index]}
        </li>
      );
    }

    listRenderer(items, ref) {
      return (
        <ul ref={ref}>
            {items}
        </ul>
      );
    }

    render() {
      return (
        <ReactList
            length={this.props.items.length}
            type={this.props.type}
            itemRenderer={this.listItemRenderer}
            itemsRenderer={this.listRenderer}
        />
      );
    }
  }

  describe('uniform react-list', () => {

    it('renders with react-test-renderer', function () {
      const tree = render(
        <MyList type={'uniform'} items={[1, 2, 3, 4, 5]} />
      );
      expect(tree).toMatchSnapshot();
    });

    it('renders into DOM', function () {
      const tree = ReactTestUtils.renderIntoDocument(
        <MyList type={'uniform'} items={[1, 2, 3, 4, 5]} />
      );
      const domNode = ReactDOM.findDOMNode(tree);
      expect(domNode).toMatchSnapshot();
    });
  });
});
