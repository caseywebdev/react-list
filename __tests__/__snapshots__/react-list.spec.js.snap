// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-list uniform react-list renders into DOM 1`] = `
<div
  data-reactroot=""
  style="position: relative;"
>
  <div>
    <ul>
      <li>
        <!-- react-text: 5 -->
        list item: 
        <!-- /react-text -->
        <!-- react-text: 6 -->
        1
        <!-- /react-text -->
      </li>
    </ul>
  </div>
</div>
`;

exports[`react-list uniform react-list renders with react-test-renderer 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTransform": "translate(0px, 0px)",
        "msTransform": "translate(0px, 0px)",
        "transform": "translate(0px, 0px)",
      }
    }
  >
    <ul>
      <li>
        list item: 
        1
      </li>
    </ul>
  </div>
</div>
`;
